android/app/src/main/java/com/expensify/chat/utils/FileUtils.kt

```diff
@@ -0,0 +1,111 @@
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+
+object FileUtils {
+    private const val tag = "FileUtils"
+    private const val directoryName = "Expensify"
+
+    private fun getInternalStorageDirectory(context: Context): File {
+        val internalStorageDirectory = File(context.filesDir.absolutePath, directoryName)
+        if (!internalStorageDirectory.exists()) {
+            internalStorageDirectory.mkdir()
+        }
+        return internalStorageDirectory
+    }
+
+    fun clearInternalStorageDirectory(context: Context) {
+        val internalStorageDirectory = getInternalStorageDirectory(context)
+        if (internalStorageDirectory.exists()) {
+            val files = internalStorageDirectory.listFiles()
+            if (files != null && files.isNotEmpty()) {
+                for (file in files) {
+                    file.delete()
+                }
+            } else {
+                Log.i(tag, "No files found to delete in directory: ${internalStorageDirectory.absolutePath}")
+            }
+        }
+    }
+
+    /**
+     * Creates a temporary file in the internal storage.
+     *
+     * @return unique file prefix
+     */
+    fun getUniqueFilePrefix(): String {
+        return System.currentTimeMillis().toString()
+    }
+
+    /**
+     * Synchronous method
+     *
+     * @param fileUri
+     * @param destinationFile
+     * @param context
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    fun saveFileFromProviderUri(fileUri: Uri?, destinationFile: File?, context: Context) {
+        val inputStream: InputStream? = fileUri?.let { context.contentResolver.openInputStream(it) }
+        val outputStream: OutputStream = FileOutputStream(destinationFile)
+        inputStream?.use { input ->
+            outputStream.use { output ->
+                input.copyTo(output)
+            }
+        }
+    }
+
+    /**
+     * Creates a temporary image file in the internal storage.
+     *
+     * @param uri
+     * @param context
+     * @return
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    fun createTemporaryFile(uri: Uri?, context: Context): File {
+
+        val mimeTypeMap = MimeTypeMap.getSingleton()
+
+        val fileExtension = uri?.let {
+            ".${mimeTypeMap.getExtensionFromMimeType(context.contentResolver.getType(it))}"
+        } ?: throw IllegalArgumentException("URI must not be null")
+
+        val file: File = File.createTempFile(
+            getUniqueFilePrefix(),
+            fileExtension,
+            getInternalStorageDirectory(context)
+        )
+
+        Log.i(tag, "Created a temporary file at" + file.absolutePath)
+        return file
+    }
+
+    /**
+     * Copy the given Uri to storage
+     *
+     * @param uri
+     * @param context
+     * @return The absolute path of the image
+     */
+    fun copyUriToStorage(uri: Uri?, context: Context): String? {
```

Comment:
I think the `Uri` doesn't have to be optional here, because in `handleSingleFileIntent` the lambda won't run unless `fileUri` exists. The same applies to other methods with `Uri?` like `createTemporaryFile` and `saveFileFromProviderUri` - I believe it will make the code much simpler, because we won't have to handle the `null` scenario in any of them 😄